src/prisma/generator/models.py

```diff
@@ -748,6 +758,13 @@ def instance_name(self) -> str:
 
         `User` -> `Prisma().user`
         """
+        config = get_config()
+        if isinstance(config, Config) and config.client_casing == ClientCasing.camel_case:
+            return to_camel_case(self.name)
+        elif isinstance(config, Config) and config.client_casing == ClientCasing.pascal_case:
+            return to_pascal_case(self.name)
+        elif isinstance(config, Config) and config.client_casing == ClientCasing.snake_case:
+            return to_snake_case(self.name)
         return self.name.lower()
```

Comment:
nit: it would be great for this to exhaustively match the enum, you can do this by using our `assert_never()` helper function. Here's an example from another part of the codebase
```py
        if value == EngineType.binary:
            return value
        elif value == EngineType.dataproxy:  # pragma: no cover
            raise ValueError('Prisma Client Python does not support the Prisma Data Proxy yet.')
        elif value == EngineType.library:  # pragma: no cover
            raise ValueError('Prisma Client Python does not support native engine bindings yet.')
        else:  # pragma: no cover
            assert_never(value)
```